src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -243,6 +253,15 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             if (mode == Mode.GlobbedHref || mode == Mode.Fallback && !string.IsNullOrEmpty(HrefInclude))
             {
+                if (TryResolveUrl(HrefInclude, tryEncodeApplicationPath: false, resolvedUrl: out resolvedUrl))
```

Comment:
Not correct for any globbed pattern. Should instead call `TryResolveUrl()` within the loops processing what `GlobbingUrlBuilder.BuildUrlList()` returns.